src/cargo/util/progress.rs

```diff
@@ -68,25 +93,30 @@ impl<'cfg> State<'cfg> {
         // 2. If we've drawn something, then we rate limit ourselves to only
         //    draw to the console every so often. Currently there's a 100ms
         //    delay between updates.
-        if self.first {
-            let delay = Duration::from_millis(500);
-            if self.last_update.elapsed() < delay {
-                return Ok(());
-            }
-            self.first = false;
-        } else {
-            let interval = Duration::from_millis(100);
-            if self.last_update.elapsed() < interval {
-                return Ok(());
+        if throttle {
+            if self.first {
+                let delay = Duration::from_millis(500);
+                if self.last_update.elapsed() < delay {
+                    return Ok(());
+                }
+                self.first = false;
+            } else {
+                let interval = Duration::from_millis(100);
+                if self.last_update.elapsed() < interval {
+                    return Ok(());
+                }
             }
+            self.last_update = Instant::now();
         }
-        self.last_update = Instant::now();
 
         // Render the percentage at the far right and then figure how long the
         // progress bar is
         let pct = (cur as f64) / (max as f64);
         let pct = if !pct.is_finite() { 0.0 } else { pct };
-        let stats = format!(" {:6.02}%", pct * 100.0);
+        let stats = match self.style {
+            ProgressStyle::Percentage => format!(" {:6.02}%", pct * 100.0),
```

Comment:
Nit: let’s move pct calculation inside the match arm to minimize scope?

Reply:
@matklad The `pct` is later used to calculate the size of the progress bar, so no.